x-pack/plugins/transform/server/routes/api/transforms_audit_messages.ts

```diff
@@ -22,10 +22,22 @@ interface BoolQuery {
 }
 
 export function registerTransformsAuditMessagesRoutes({ router, license }: RouteDependencies) {
-  router.get(
-    { path: addBasePath('transforms/{transformId}/messages'), validate: schemaTransformId },
+  /**
+   * @apiGroup Transforms Audit Messages
+   *
+   * @api {get} /api/transform/transforms/:transformId/messages Transforms Messages
+   * @apiName GetTransformsMessages
+   * @apiDescription Get transforms audit messages
+   *
+   * @apiSchema (params) transformIdParamSchema
+   */
+  router.get<TransformIdParamSchema, undefined, undefined>(
+    {
+      path: addBasePath('transforms/{transformId}/messages'),
+      validate: { params: transformIdParamSchema },
+    },
     license.guardApiRoute(async (ctx, req, res) => {
-      const { transformId } = req.params as SchemaTransformId;
+      const { transformId } = req.params as TransformIdParamSchema;
```

Comment:
`guardApiRoute` could be changed accept the `params` type.

```
  guardApiRoute<T = unknown>(handler: RequestHandler<T, unknown, any, any>) {
    const license = this;

    return function licenseCheck(
      ctx: RequestHandlerContext,
      request: KibanaRequest<T>,
      response: KibanaResponseFactory
    ): IKibanaResponse<any> | Promise<IKibanaResponse<any>> {
      ....
```

and then called with:
```
license.guardApiRoute<IndexPatternTitleSchema>(async (ctx, req, res) => {
      const { indexPatternTitle } = req.params;
```

It would be even better to have an implicit `Typeof` conversion of the `validate` value, but i don't think that is possible.

Reply:
I was able to do some improvement in cfe297f.